pyanaconda/packaging/dnfpayload.py

```diff
@@ -797,23 +795,10 @@ def preInstall(self, packages=None, groups=None):
             self.requiredPackages += packages
         self.requiredGroups = groups
 
-        # Write the langpacks config
-        pyanaconda.iutil.mkdirChain(DNF_PLUGINCONF_DIR)
-        langs = [self.data.lang.lang] + self.data.lang.addsupport
-
-        # Start with the file in /etc, if one exists. Otherwise make an empty config
-        if os.path.exists(_DNF_TARGET_LANGPACK_CONF):
-            shutil.copy2(_DNF_TARGET_LANGPACK_CONF, _DNF_INSTALLER_LANGPACK_CONF)
-        else:
-            with open(_DNF_INSTALLER_LANGPACK_CONF, "w") as f:
-                f.write("[main]\n")
-
-        # langpacks.conf is an INI style config file, read it and
-        # add or change the enabled and langpack_locales entries without
-        # changing anything else.
-        keys=[("langpack_locales", "langpack_locales=" + ", ".join(langs)),
-              ("enabled", "enabled=1")]
-        simple_replace(_DNF_INSTALLER_LANGPACK_CONF, keys)
+        # add langpacks into transaction
+        for lang in [self.data.lang.lang] + self.data.lang.addsupport:
```

Comment:
A little bit extra effort needs to be spent here. Both ``self.data.lang.lang`` and ``self.data.lang.addsupport`` contain things like ``en_US.UTF-8``, ``cs_CZ.UTF-8`` and so on. ``pyanaconda.localization.parse_langcode()`` needs to be used here to separate the langcode.